doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+
+As a consequence of this log-like data structure, Holochain is a departure from traditional relational
+(SQL) and also non-relational (noSQL) databases. There are no tables, and no foreign keys, at least on the formal level of Holochain.
+Despite this, one must of course still be able to persist data, and read back persisted data, and all of that happens in relation to the local source chain.
+
+With Holochain, there are two aspects that combine to form each "element" in this chain-like log: headers, and entries.
+
+### Headers
+
+The fundamental chain-like aspect of Holochain is driven by "headers". They are a minimal data structure containing only a handful of values, but are sufficient to supply Holochain with verifiable data integrity.
+
+Fundamental to creating the "chain", each header contains a reference to the "last" header. There are a couple aspects to this:
+1. The cryptographic hash (its data "fingerprint") for a header is determined
+2. The hash of the previously written header is used as a fixed-length input for the next header, thus influencing the hash of that new header (since any change to contents change a hash)
+3. The only time this does not apply is for the very first header, for which there is no previous header hash, and a value of `0` is used
+
+Other data contained in the headers includes (but isn't limited to):
+- the public key/address, and the cryptographically signed signature of the author
+- a timestamp conveying the moment of creation of the header
+- the hash of the "entry" being created
+- the "entry type" of the "entry" being created
+
+The meaning and role of entries is covered next.
+
+### Entries
+
+Entries are elements of data which get persisted to the same storage as Headers, yet have a different nature. They represent the substance of the data that is of interest to the end users of Holochain, storing records such as transactions, social media posts, or whatever else app developers come up with. The hash of each entry is stored and referenced in its corresponding header. Every local source chain should have the exact same number of Entries and Headers.
+
+All Entries have a type. Entries can be "system" Entries, or "app" Entries. System Entries are Entries integral to the proper functioning of Holochain. Importantly, there are system Entries written to each local source chain, at the time of its initialization, or "genesis".
+
+#### System Entries
+These include firstly, the DNA itself. The DNA is written as the first entry into each local source chain. Importantly, this is what gives everyone totaly confidence in everyone elses chains: the fact that you all know you started out by running the same code (otherwise, the hash of the initial DNA entry would be different). There will be more discussion on this later.
```

Comment:
```suggestion
These include firstly, the DNA itself. The DNA is written as the first entry into each local source chain. Importantly, this is the foundation for establishing confidence in the data of everyone's chains: the fact that you all know you started out by running the same code (otherwise, the hash of the initial DNA entry would be different). There will be more discussion on this later.
```

Given that nobody knows whether anyone else is actually running the DNA they said they're running, I thought we ought to reframe it as simply setting the baseline for confidence -- "anyone who claims to be running this DNA is going to be held to the rules of this DNA"

Reply:
nice, yeah